From fc5a069e4bba34cc52df04a39b423d3dfdec5da0 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Wed, 5 Aug 2015 20:13:22 +0200
Subject: [PATCH 1/3] video: mxc_edid: Parse Video Capability Data Block

The information from this block is used to detect if a
sink supports overriding of the RGB quantisation range.
---
 drivers/video/mxc/mxc_edid.c |   16 ++++++++++++++++
 include/video/mxc_edid.h     |    4 ++++
 2 files changed, 20 insertions(+)

diff --git a/drivers/video/mxc/mxc_edid.c b/drivers/video/mxc/mxc_edid.c
index 88b5268..6e2b04c 100644
--- a/drivers/video/mxc/mxc_edid.c
+++ b/drivers/video/mxc/mxc_edid.c
@@ -547,6 +547,22 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 					break;
 				}
 			case 0x7: /*User extended block*/
+				if (blklen >= 2 && edid[index + 1] == 0) {	/*Video Capability Data Block*/
+					u8 data = edid[index + 2];
+
+					cfg->cea_scan_mode_ce = (data >> 0) & 3;
+					cfg->cea_scan_mode_it = (data >> 2) & 3;
+					cfg->cea_scan_mode_pt = (data >> 4) & 3;
+					cfg->cea_rgb_range_selectable = (data >> 6) & 1;
+
+					DPRINTK("VCDB over/underscan behavior (CE)  %d\n", cfg->cea_scan_mode_ce);
+					DPRINTK("VCDB over/underscan behavior (IT)  %d\n", cfg->cea_scan_mode_it);
+					DPRINTK("VCDB over/underscan behavior (PT)  %d\n", cfg->cea_scan_mode_pt);
+					DPRINTK("VCDB RGB quant. range selectable   %d\n", cfg->cea_rgb_range_selectable);
+
+					index += blklen;
+					break;
+				}
 			default:
 				/* skip */
 				DPRINTK("Not handle block, tagcode = 0x%x\n", tagcode);
diff --git a/include/video/mxc_edid.h b/include/video/mxc_edid.h
index 34e797c..9d34f22 100644
--- a/include/video/mxc_edid.h
+++ b/include/video/mxc_edid.h
@@ -64,6 +64,10 @@ struct mxc_edid_cfg {
 	bool cea_ycbcr444;
 	bool cea_ycbcr422;
 	bool hdmi_cap;
+	bool cea_rgb_range_selectable;
+	u8 cea_scan_mode_ce;
+	u8 cea_scan_mode_it;
+	u8 cea_scan_mode_pt;
 
 	/*VSD*/
 	bool vsd_support_ai;
-- 
1.7.9.5

